.github/workflows/auto_release.yml

```diff
@@ -0,0 +1,58 @@
+name: Auto-release
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  check_version:
+    name: Check the package version
+    runs-on: ubuntu-latest
+    outputs:
+      version_changed: ${{ steps.check_result.outputs.version_changed }}
+      new_version: ${{ steps.check_result.outputs.new_version }}
+    steps:
+      - name: Check out the code
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+      - name: Check the package version
+        id: check_result
+        run: |
+          latest_release_version=$(git for-each-ref --sort=-creatordate --count 1 --format="%(refname:short)" "refs/tags/*")

```

Comment:
Does this require us to push a new tag as well? I always just do the "generate a new tag on release publish" option

Reply:
No, you just create a version bump commit and that's it

Reply:
Ah yes, this is getting the latest current tag, I see it. I'm tired 🙃